src/Domain.jl

```diff
@@ -1632,6 +1632,20 @@ end
     end
 end
 
+@inline function jacobiany_ns!(jac::AbstractArray,domain::Union{ConstantPDomain,ParametrizedPDomain},rxnarray::Array{Int64,2},efficiencyinds::Array{I,1},cs::Array{Float64,1},kfs::Array{Float64,1},krevs::Array{Float64,1},T::Float64,V::Float64,C::Float64) where {I<:Integer}
+    @simd for rxnind = 1:size(rxnarray)[2]
+        _calc_jacobiany_ns_deriv_ns!(jac,rxnarray[1:3,rxnind],rxnarray[4:6,rxnind],cs,kfs[rxnind])
```

Comment:
So I chose to pass the rxnarray separately here so that I can reuse _calc_jacobiany_ns_deriv_ns! for both reactant direction and product direction, and it also helps to make  _calc_jacobiany_ns_deriv_ns!  more readable (Instead of having lots of rxnarray slicing inside the function, we only have the slicing at the function input). I will add **@views** !

Reply:
I tested two version of jacobiany_ns!, one uses slicing with **@views**, and the other avoids all these slicing, and I found that the latter one is slightly faster (1 second) for PME mechanism ConstantTPDomain simulation. I will change all of the relevant functions to minimize the usage of slicing (even though using slicing gives clear and simpler code).